src/auth/authController.ts

```diff
@@ -13,3 +14,51 @@ export const refreshAccessToken = async (req: Request, res: Response) => {
   const tokens = await authService.refreshAccessToken(refreshToken);
   return res.json(tokens);
 };
+
+/**
+ * @desc Links a GET/CBORD account to a deviceId via a PIN.
+ * This is the one-time setup for the finance feature.
+ */
+export const getAuthorize = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  const { userId } = req.user!;
+  const { pin, sessionId } = req.body;
+
+  try {
+    await cbordService.createPin(String(userId), pin, sessionId);
+
+    // This route does NOT create the user, just links the PIN.
+    return res
+      .status(200)
+      .json({ message: 'GET account linked successfully.' });
+  } catch (error) {
+    return next(error);
```

Comment:
You shouldn't need to have this `try-catch` error checking code here as we have a global error handler that will intercept the error should it appear. Controller functions should pretty much only be parsing the incoming HTTP request and returning the result from the service.